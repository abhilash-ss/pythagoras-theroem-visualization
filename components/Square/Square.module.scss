.square {
  border: 4px solid black;

  &__row {
    display: flex;
    background: black;
  }

  &__cell {
    height: 40px;
    width: 40px;
    background: lightgrey;
    margin: 1px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 45px;
    font-weight: 800px;
    color: white;

    &--base {
      color: red;
    }

    &--altitude {
      color: blue;
    }
  }
  &__base {
    color: red;
  }
}
