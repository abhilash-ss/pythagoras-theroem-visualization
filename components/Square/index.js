import React from "react";
import styles from "./Square.module.scss";

function Square({ size, type, opacity = 1 }) {
  const row = Array.from(Array(size).keys());
  const cellColorClass =
    type === "base" ? "square__cell--base" : "square__cell--altitude";
  return (
    <div className={styles.square} style={{ opacity: opacity }}>
      {row.map((i) => {
        return (
          <div className={styles.square__row}>
            {row.map((i) => (
              <div
                className={styles.square__cell}
                style={{ background: type === "base" ? "blue" : "red" }}
              />
            ))}
          </div>
        );
      })}
      <style jsx>{`
        .container {
          margin: 50px;
        }
        p {
          color: blue;
        }
      `}</style>
    </div>
  );
}

export default Square;
