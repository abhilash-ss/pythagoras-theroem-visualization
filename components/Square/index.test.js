import React from "react";
import { shallow } from "enzyme";
import Square from "./index";

describe("Square", () => {
  let props;
  beforeEach(() => {
    props = {
      size: 5,
      type: "base"
    };
  });
  it("should render without crash", () => {
    const component = shallow(<Square {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should render without type hypotenuse", () => {
    props.type = "hypotenuse";
    const component = shallow(<Square {...props} />);
    expect(component).toMatchSnapshot();
  });
});
