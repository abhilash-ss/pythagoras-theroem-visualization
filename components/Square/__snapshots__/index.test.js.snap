// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render without crash 1`] = `ShallowWrapper {}`;

exports[`Square should render without type hypotenuse 1`] = `ShallowWrapper {}`;
